### PR TITLE
tests/scale: pod/traffic split scaling test

### PR DESCRIPTION
**Pull request by eduser25**
Body and operation of the test is close to identical to existing
`e2e_trafficsplit_test.go`, with few differences to be able to 
iterate repetitively over the body of the test itself.

Test will repeat the test operation till a failure is observed.
At each iteration, the timestamps are recorded, and later statistics
for relevant pods are gathered and output as part of the results,
for every iteration and its duration.

Test results are stored in a test folder generated by the test itself,
if not given by parametre. 

- `tests/scale/`
  - `osm_common_profile_resources.go`: contains a set of convenience
  functions that define usual resources and outputs we are interested
  in keeping/having for OSM specific tests.
  - `scale_trafficSplit_test.go`: test implementation

Please mark with X for applicable areas.

- Metrics                [X]
- Tests                  [X]
- Performance            [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No.

### CODE REVIEW

**Comment by eduser25** [comment]
Outputs examples:
```
osm/tests/scale/test-3440715768 (scale-test) $ ls
cpu.png  mem.png  results.txt
```
`results.txt`:
```
+----+---------------------------+-------+-------------------------+-------------------------+
| It |         Duration          | NPods |  osm-con..gfhb/osm-..   |  osm-pro..5x4c/prom..   |
+----+---------------------------+-------+-------------------------+-------------------------+
|  0 | 40.968050138s             |    20 | 0.22 / 43 MB            | 0.06 / 215 MB           |
|  1 | 32.205140695s (-21.39%)   |    40 | 0.24 / 57 MB (+32.23%)  | err / 263 MB (+22.67%)  |
|  2 | 37.255469213s (+15.68%)   |    60 | 0.61 / 70 MB (+23.85%)  | 0.10 / 380 MB (+44.53%) |
|  3 | 40.251252794s (+8.04%)    |    80 | 0.70 / 84 MB (+19.26%)  | 0.14 / 504 MB (+32.52%) |
|  4 | 38.200125109s (-5.10%)    |   100 | 1.02 / 95 MB (+13.57%)  | 0.13 / 593 MB (+17.54%) |
|  5 | 46.80032277s (+22.51%)    |   120 | 1.22 / 110 MB (+16.35%) | 0.12 / 742 MB (+25.28%) |
|  6 | 54.901534208s (+17.31%)   |   140 | 1.30 / 126 MB (+13.73%) | 0.17 / 833 MB (+12.22%) |
|  7 | 1m3.895762229s (+16.38%)  |   160 | 1.21 / 146 MB (+16.06%) | 0.20 / 1.0 GB (+21.42%) |
|  8 | 1m18.193766557s (+22.38%) |   180 | 1.41 / 161 MB (+10.61%) | 0.18 / 1.1 GB (+8.23%)  |
|  9 | 1m38.286988829s (+25.70%) |   200 | 1.33 / 179 MB (+11.07%) | 0.19 / 1.2 GB (+14.12%) |
| 10 | 1m51.925769047s (+13.88%) |   220 | 1.43 / 202 MB (+12.44%) | 0.21 / 1.4 GB (+12.40%) |
| 11 | 2m20.548972682s (+25.57%) |   240 | 1.48 / 219 MB (+8.66%)  | 0.24 / 1.5 GB (+10.25%) |
+----+---------------------------+-------+-------------------------+-------------------------+
+----------------------+-----------+
|       Resource       | Mem / pod |
+----------------------+-----------+
| osm-con..gfhb/osm-.. | 801 kB    |
| osm-pro..5x4c/prom.. | 6.1 MB    |
+----------------------+-----------+```
`cpu.png`:
![cpu](https://user-images.githubusercontent.com/5503924/101844710-eb159b80-3b01-11eb-83f3-ae4afadf4c0e.png)
`mem.png`:
![mem](https://user-images.githubusercontent.com/5503924/101844726-f36dd680-3b01-11eb-8b36-a8525dae1ac7.png)

File: tests/scale/scale_trafficSplit_test.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+				Expect(Td.AddNsToMesh(true, allNamespaces...)).To(Succeed())
+
+				// Create server services
+				for _, serverApp := range serverServices {
+					svcAccDef, deploymentDef, svcDef := Td.SimpleDeploymentApp(
+						SimpleDeploymentAppDef{
+							Name:         serverApp,
+							Namespace:    serverNamespace,
+							ReplicaCount: int32(serverReplicaSet),
+							Image:        "simonkowallik/httpbin",
```
Comment (nojnhuh) [inline tests/scale/scale_trafficSplit_test.go:100]: The image we use in most other tests is `kennethreitz/httpbin`. Probably worth settling on one.

**Review by nojnhuh — DISMISSED** [review]
lgtm, a couple nits

File: tests/scale/scale_trafficSplit_test.go (hunk 1 above)
Comment (eduser25) [reply tests/scale/scale_trafficSplit_test.go:100]: I think original httpbin didn't have the `env to header` feature we are using to verify the pod which answered a request...
https://github.com/simonkowallik/httpbin

**Review by shashankram — APPROVED** [review]

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**